app/auth/profile/deleteprofile.tsx

```diff
@@ -0,0 +1,144 @@
+import React, { useState } from "react";
+import { zodResolver } from "@hookform/resolvers/zod";
+import { useForm } from "react-hook-form";
+import { router } from "expo-router";
+import { z } from "zod";
+import { View, StyleSheet, Text, TouchableOpacity, Keyboard, Alert } from "react-native";
+import FormContainer from "../../../components/forms/FormContainer";
+import FormHeader from "../../../components/forms/FormHeader";
+import FormField from "../../../components/forms/TextInput";
+import SubmitButton from "../../../components/forms/SubmitButton";
+import useProfile from "../../../hooks/useProfile";
+import { useAuthentication } from "../../../providers/AuthenticationProvider";
+import { useToast } from "../../../providers/ToastProvider";
+import { colors, ScaleSize, ScaleSizeH, ScaleSizeW } from "../../../utils/SharedStyles";
+
+const schema = z
+  .object({
+    currentPassword: z.string().trim().min(8, "Indtast nuværende adgangskode"),
+    confirmPassword: z.string().trim(),
+  })
+  .superRefine((data, ctx) => {
+    if (data.currentPassword !== data.confirmPassword) {
+      ctx.addIssue({
+        code: z.ZodIssueCode.custom,
+        path: ["confirmPassword"],
+        message: "Adgangskode stemmer ikke overens",
+      });
+    }
+  });
+
+type FormData = z.infer<typeof schema>;
+
+const DeleteProfileScreen: React.FC = () => {
+  const [password, setPassword] = useState("");
+  const { addToast } = useToast();
+  const { logout } = useAuthentication();
+  const { userId } = useAuthentication();
```

Comment:
Du behøver ikke importe dem her to gange. Det er muligt at skrive const { logout, userId } = useAuthentication(); :))